ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -387,13 +400,15 @@ withSnocket
        )
     => Tracer m (WithAddr (TestAddress peerAddr)
                           (SnocketTrace m (TestAddress peerAddr)))
-    -> Script BearerInfo
+    -> BearerInfo
+    -> Map (NormalisedId (TestAddress peerAddr))
+          (LazySTM.TVar m (Script BearerInfo))
```

Comment:
I thought there was but there isn't, I can just update the entry. I'll update that!

Reply:
Actually it is easier to keep things as they are because the alternative would require to make the whole map a TVar. Unless you prefer it

Reply:
Are you sure:
```
f :: Map a b -> IO (Map a (TVar b))
f = traverse newTVarIO
```

Reply:
Yes because `Snocket` gets passed the network state and the actions (`accept`, `connect`, etc..) all reach for the state, so we need thing in `TVar` so we can update some of the structures in place. 

I use the pure version in the generators and` traverse newTVarIO` is what I use when initializing the state: https://github.com/input-output-hk/ouroboros-network/pull/3429/files#diff-c14d3f02791c85daaf52c0549d7323e19d21e6055d7d921276460786771d9131R3104

Reply:
You just need to pass the `Map ... (TVar )`. to `accept`, `connect` etc as an argument, that's fine.

Reply:
I will!

Reply:
@coot Actually, I thought you wanted to remove the TVars from the NetworkState definition but now I realize it might be in `newNetworkState`, if so that makes more sense. The pure version already comes from the outside. In `newNetworkState` we just bundle everything together so I think it is better to just switch the type signature from that function to everything consistent.

Reply:
Fixed!